### PR TITLE
Implement Array.of method

### PR DESCRIPTION
**Pull request by szilagyiadam**
The algorithm is based on ECMA-262 v6, 22.1.2.3

JerryScript-DCO-1.0-Signed-off-by: Adam Szilagyi aszilagy@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-array.c [hunk 1]
```diff
@@ -390,6 +390,79 @@ ecma_builtin_array_object_from (ecma_value_t this_arg, /**< 'this' argument */
...
+                                          ECMA_VALUE_UNDEFINED,
+                                          &len,
+                                          1);
+  }
+  else
+  {
+    ret_val = ecma_op_create_array_object (&len, 1, true);
+  }
+
+
```
Comment (dbatyai) [inline jerry-core/ecma/builtin-objects/ecma-builtin-array.c]: Extra newline

**Comment by zherczeg** [comment]
I think we havbe a constructor which constructs an array from a list of elements. Can't we use it?

**Comment by rerobika** [comment]
@zherczeg The constructor always creates an array instance, however this method is generic.

**Comment by dbatyai** [comment]
@rerobika, @zherczeg We could do something like:
```
if (!ecma_is_constructor (this_arg))
{
  return ecma_op_create_array_object (arguments_list_p, arguments_list_len, false);
}

/* Do generic construct. */
```

**Comment by szilagyiadam** [comment]
I did the requested changes above.

**Pull request merged**